lib/dom.js

```diff
@@ -3568,6 +3568,36 @@ declare class ClientRectList { // extension
   [index: number]: ClientRect;
 }
 
+declare class DOMRectReadOnly {
+  +bottom: number;
+  +height: number;
+  +left: number;
+  +right: number;
+  +top: number;
+  +width: number;
+  +x: number;
+  +y: number;
+  fromRect(rectangle?: {x: number, y: number, width: number, height: number}): DOMRect;
```

Comment:
couple of things:

1) should be `static`
2) if i'm reading [the draft](https://drafts.fxtf.org/geometry/#domrectreadonly) correctly, `DOMRectReadOnly.fromRect()` should return a `DOMRectReadOnly`, and `DOMRect.fromRect()` should return a `DOMRect`. so add one to DOMRect as well